pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -548,6 +557,65 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 	return nil
 }
 
+func (r *ReconcileSyncSetInstance) reconcileDeletedSyncSetResources(applyMode hivev1.SyncSetResourceApplyMode, dynamicClient dynamic.Interface, existingStatusList, newStatusList []hivev1.SyncStatus, err error, ssiLog log.FieldLogger) ([]hivev1.SyncStatus, error) {
+	ssiLog.Debugf("reconciling syncset resources, existing: %d, actual: %d", len(existingStatusList), len(newStatusList))
+	if applyMode == "" || applyMode == hivev1.UpsertResourceApplyMode {
+		ssiLog.Debugf("apply mode is upsert, syncset status will be updated")
+		return newStatusList, nil
+	}
+	deletedStatusList := []hivev1.SyncStatus{}
+	deletedStatusIndices := []int{}
+	for i, existingStatus := range existingStatusList {
+		found := false
+		for _, newStatus := range newStatusList {
+			if existingStatus.Name == newStatus.Name &&
+				existingStatus.Namespace == newStatus.Namespace &&
+				existingStatus.APIVersion == newStatus.APIVersion &&
+				existingStatus.Kind == newStatus.Kind {
+				found = true
+				break
+			}
+		}
+		if !found {
+			ssiLog.WithField("resource", fmt.Sprintf("%s/%s", existingStatus.Namespace, existingStatus.Name)).
+				WithField("apiversion", existingStatus.APIVersion).
+				WithField("kind", existingStatus.Kind).Debug("resource not found in updated status, will queue up for deletion")
+			deletedStatusList = append(deletedStatusList, existingStatus)
+			deletedStatusIndices = append(deletedStatusIndices, i)
+		}
+	}
+
+	// If an error occurred applying resources, do not delete yet
+	if err != nil {
```

Comment:
Along with continuing to apply other secrets after a failure, we should attempt to delete secrets even if there was an error during an apply.